### ISSUE TITLE
Equatable should override toString and return a String with the name of each property and value

### ISSUE THREAD

**Issue by GiancarloCode** [felangel/equatable#27]
**Is your feature request related to a problem? Please describe.**

If you have a `Credential` class:

```dart
class Credentials extends Equatable {
  final String username;
  final String password;

  Credentials({this.username, this.password}) : super([username, password]);
}
```
When you create a credential and print:
```dart
final credentials = Credentials(username: 'Joe', password: 'password123');
print(credentials)
```
The output is:
```shell
Instance of 'Credentials'
```
In most cases you want to show the name of class, the property name and the value of each property, so you need to override `toString()` manually:

```dart
class Credentials extends Equatable {
  final String username;
  final String password;

  Credentials({this.username, this.password}) : super([username, password]);

  @override
  String toString() =>
      'Credentials { userName: $username, password: $password }';
}
```

And now the output is:
```shell
Credentials { userName: Joe, password: password123 }
```

**Describe the solution you'd like**

Equatable accept a map instead of list.

```dart
import 'package:meta/meta.dart';
import './equatable_utils.dart';

@immutable
abstract class Equatable {
  final Map props;

  Equatable([this.props = const {}]);

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Equatable &&
          runtimeType == other.runtimeType &&
          equals(props.values.toList(), other.props.values.toList());

  @override
  int get hashCode =>
      runtimeType.hashCode ^ mapPropsToHashCode(props.values.toList());

  @override
  String toString() => '$runtimeType${_propsToString()}';

  String _propsToString() {
    StringBuffer _result = StringBuffer();

    if (props.isNotEmpty) {
      _result.write(' {');
      for (int i = 0; i < props.length; i++) {
        _result
          ..write(' ')
          ..write(props.keys.elementAt(i))
          ..write(': ')
          ..write(props.values.elementAt(i));
        if (i != props.length - 1) _result..write(',');
      }
      _result.write(' }');
    }

    return _result.toString();
  }
}

```
Now you need to pass a `Map`  to the super constructor, when the key is the String for show in `toString()` method, and the value, is the value to use to override `==` and `hashcode`
```dart
class Credentials extends Equatable {
  final String username;
  final String password;

  Credentials({this.username, this.password})
      : super({'userName': username, 'password': password});
}
```
And now the output is:
```shell
Credentials { userName: Joe, password: password123 }
```
Thanks 😀

**Comment by felangel** [comment]
@GiancarloCode thanks for the feature request!

Based on our discussion on gitter, I am going to update `toString` to default to `runtimeType` because passing a map increases the complexity and is also tangential to the focus of the package.

**Comment by felangel** [comment]
Published in v0.4.0 of equatable 🎉
